### ISSUE TITLE
8FaceMK2 will crash VCV if using more than 1 module or its +8 extension

### ISSUE THREAD

**Issue by jaynothin** [stoermelder/vcvrack-packone#396]
There is a "hiccup" when using 8FaceMK2 that seem to freeze for a fraction of a sec the graphic. It happens like every 10sec (didn't really count ;) ).
If you add more 8Face or its extension the +8 module, the freeze gets worst, to the point of crashing VCV.

I've made a patch to share so it's easy to reproduce. [BugStoermelder8Face+8.zip](https://github.com/user-attachments/files/17535752/BugStoermelder8Face%2B8.zip)
The patch has been created to make VCV crash on my computer.
But if it doesn't crash, you should be able to see the "hiccup/freeze" of the graphic.

Before reporting the issue I've asked someone on a different computer, more recent and powerful than mine, to try the patch and it made his VCV crash after a few "freeze".

my old machine:
i7 4770
24G RAM
RX 470 ASUS ROG Strix
Win10 latest
VCV Pro 2.5.2

**Comment by stoermelder** [comment]
I'm getting a virus warning for the zip file. Is there anything else included in the archive or just a patch file?

**Comment by jaynothin** [comment]
It's only the patch, dunno why there is a warning? 
🤔

**Comment by jaynothin** [comment]
In case my zip is rejected I'm going to leave you a PM in the VCV forum with the file .vcv not zipped.

**Comment by stoermelder** [comment]
I spent the last two hours investigating this issue and I'm not sure what is causing this. It seems it has something to do with the trowaSoft TrigSeq64, as I can reproduce it only if both modules are in the patch. But I'm not sure which one is responsible.
Can you check if it also happens when you don't use a trowaSoft module in your patch?

**Comment by jaynothin** [comment]
Yes, in fact just couple hours ago earlier I decided to use something else. 
I only had the mindmeld mixer, and each slots of the 8FACEmk2 plus 4-5 +8 modules were set to a different random mixer settings. I plugged in a clock and it was stable, but probably like 60 sec later, VCV crashed and closed on it's own. 

So this took TrowaSoft out of the way of being possibly the problem. :)

**Comment by stoermelder** [comment]
I did take a quick look at trowaSoft's code and I don't think it is caused by anything these modules are doing. I'm not sure right now what is causing this.
I will investigate further. Does is also happen with classic 8FACE (mk1)?

**Comment by jaynothin** [comment]
I was wonderinf this : 

Since I'm seeing the "freeze" of the GUI every 10/15 seconds, I was thinking if it has anything to do with VCV auto-saving the patch?

I've made a few patch recently only using the 8Face (mk1)
So far I didn't get a crash, but I do see the "freeze" happening. 

That's what made me wonder about the auto save frequency ( 10 ?? 20 ?? second) but I don't knwo about this.

**Comment by stoermelder** [comment]
Yes, it is happening while auto-saving the patch, your observation is correct.

**Comment by stoermelder** [comment]
I started investigating this issue again:
- The patch file you provided is about 80MB big. As vcv files are compressed, you can't see this immediately, but loading the file will decompress it into the `autosave` folder, where you can observe it.
- Preset files of trowaSoft TrigSeq64 are pretty big on their own, ca. 700kb for an empty instance. This seems right, because 64 patterns x 16 channels x 64 steps, but creates crazy amount of data.
- The "hiccup" is caused by the size of the patch file. Unfortunately this can't be avoided, except trowaSoft finds a way to optimize the preset size. In the meantime you shouldn't use 8FACE (mk2) for this module.
- Additionally there was a bug in 8FACE mk2: In your preset, 8FACE mk2 holds presets for an instance trowaSoft TrigSeq64, which is not in the patch anymore (see "[ERROR] ..." in the context menu). "Unbind" should clear these presets, but this was broken. With the latest build the orphaned presets get cleared correctly and the patch size returns to a reasonable size.

<img width="968" height="576" alt="Image" src="https://github.com/user-attachments/assets/2b62790b-4791-413a-b667-8e4227900990" />
